hawc/apps/epiv2/mixins.py

```diff
@@ -0,0 +1,69 @@
+from rest_framework import exceptions, serializers
+
+from .models import Design
+
+
+class BelongsToSameDesignMixin:
+    """
+    ex: when updating an ExposureLevel, there's a related Chemical and Exposure sometimes passed into the payload as
+    "chemical_id" and "exposure_level_id". Basic DRF functionality verifies that the supplied values are valid
+    items in the database, but it's a common requirement that they also belong to the same parent object; in this
+    case, the same design. This mixin performs that type of validation logic.
+
+    Could potentially generalize this even further (so that it's not just useful for "Design"); this type of
+    relationship is not uncommon.
+    """
+
+    def validate(self, data):
+        fields_to_check = []
+        try:
+            fields_to_check = self.belongs_to_same_design_fields
+
+            # first off - were any of the parameters we're concerned with in the payload? If not, there's
+            # no need to perform any custom validation.
+            perform_custom_validation = False
+            for field_to_check in fields_to_check:
+                param_name = field_to_check[0]
+                if param_name in self.initial_data:
+                    perform_custom_validation = True
+                    break
+
+            if perform_custom_validation is True:
+                # figure out what design we're concerned with...
+                if "design" in self.initial_data:
+                    # load from the supplied payload...
+                    design_id = self.initial_data.get("design")
+                    design = Design.objects.get(id=design_id)
+                else:
+                    # load from the previously saved state; the APIException here should never happen but just in case...
+                    if self.instance is not None:
+                        design = self.instance.design
+                    else:
+                        raise exceptions.APIException(
```

Comment:
maybe rewrite?; I'd probably put all this logic in another method for named `_get_design` or something

```python

design = getattr(self.instance, 'design', None) if self.instance else None
design_id = self.initial_data.get("design")
if design_id:
    try:
        design = Design.objects.get(id=design_id)
    except not found  # catch if id doesnt exist
        design = None
if design is None:
    raise ValidationError()
return design
```

Reply:
done, broke it out into funtion _get_contextual_design in the mixin, and reworked it a bit